### PR TITLE
Refactor condition opcodes and add CMOVcc and SETcc

### PR DESCRIPTION
**Pull request by Earlz**

### CODE REVIEW

**Comment by Earlz** [comment]
DO NOT MERGE.. More work being done

File: src/decoding.rs [hunk 1]
```diff
@@ -168,7 +168,7 @@ pub struct ParsedModRM{
 impl ParsedModRM{
     /// Decodes the current byte stream (starting at the ModRM byte) into a fully formed ParsedModRM struct
     pub fn from_bytes(bytestream: &[u8]) -> Result<ParsedModRM, VMError>{
-        if bytestream.len() < 16{
+        if bytestream.len() < 5{
```
Comment (VoR0220) [inline src/decoding.rs]: why disconnect between this and the len change below?

File: src/opcodes.rs [hunk 2]
```diff
@@ -487,166 +516,17 @@ lazy_static! {
...
             .into_table(&mut ops);
-        //0x70 JO rel8
-        define_opcode(0x70).calls(jmp_overflow).with_gas(Low)
-            .with_arg(ArgSource::JumpRel, Fixed(Byte))
-            .is_unpredictable()
-            .into_table(&mut ops);
-        //0x71 JNO rel8
-        define_opcode(0x71).calls(jmp_not_overflow).with_gas(Low)
+        //0x70-0x7F Jcc rel8
+        define_opcode_multi(0x70, 16).calls(jcc).with_gas(Low)
```
Comment (VoR0220) [inline src/opcodes.rs]: nice

File: src/opcodes.rs [hunk 3]
```diff
@@ -487,166 +516,17 @@ lazy_static! {
...
-            .into_table(&mut ops);
-        //0x8E JLE relw
-        define_opcode(0x8E).is_two_byte_op().calls(jmp_less_or_equal).with_gas(Low)
-            .with_arg(ArgSource::JumpRel, NativeWord)
-            .is_unpredictable()
-            .into_table(&mut ops);
-        //0x8F JG relw
-        define_opcode(0x8F).is_two_byte_op().calls(jmp_greater).with_gas(Low)
+        //0x80-0x8F Jcc relw
+        define_opcode_multi(0x80, 16).is_two_byte_op().calls(jcc).with_gas(Low)
```
Comment (VoR0220) [inline src/opcodes.rs]: nice

File: src/decoding.rs (hunk 1 above)
Comment (Earlz) [reply src/decoding.rs]: The Mod R/M data can be at most 5 bytes large, while the decode_args_with_modrm stuff also includes the opcode byte, so can be at most 6 bytes large in that context

**Pull request merged**